extensions/ruby/src/ruby.rs

```diff
@@ -0,0 +1,62 @@
+mod language_servers;
+
+use zed::lsp::{Completion, Symbol};
+use zed::{CodeLabel, LanguageServerId};
+use zed_extension_api::{self as zed, Result};
+
+use crate::language_servers::Solargraph;
+
+struct RubyExtension {
+    solargraph: Option<Solargraph>,
+}
+
+impl zed::Extension for RubyExtension {
+    fn new() -> Self {
+        Self { solargraph: None }
+    }
+
+    fn language_server_command(
+        &mut self,
+        language_server_id: &LanguageServerId,
+        worktree: &zed::Worktree,
+    ) -> Result<zed::Command> {
+        match language_server_id.as_ref() {
+            Solargraph::LANGUAGE_SERVER_ID => {
+                let solargraph = self.solargraph.get_or_insert_with(|| Solargraph::new());
+
+                Ok(zed::Command {
+                    command: solargraph.server_script_path(worktree)?,
+                    args: vec!["stdio".into()],
+                    env: worktree.shell_env(),
```

Comment:
To pick up the correct Ruby version for a project, version managers like `asdf`, `mise` or `rbenv` require to use shell environment.